### PR TITLE
Vet clinic database: add "join table" for visits

### PR DESCRIPTION
**Pull request by ticoniq**
**Vet Clinic Database: Add "Join Table" for Visits**

This pull request introduces a crucial enhancement to the Vet Clinic database by implementing a "join table" to handle the relationship between animals and vets during visits. The new `visits` table provides a structured way to track animal visits to vets, including the date of the visit.

**Changes Made:**

1. Created the `visits` table to manage the many-to-many relationship between animals and vets during visits.
2. Added columns to store `animal_id`, `vet_id`, and `visit_date`.
3. Included an auto-incremented `id` column as the primary key for each visit record.

This update adds a layer of organization to the data, making it easier to manage and retrieve information about animal visits. Screenshots of the new `visits` table and sample queries have been included in the pull request for verification.

1 How many different animals did Stephanie Mendez see?
Screenshot 2023-08-17 065734

2 List all vets and their specialties, including vets with no specialties.
Screenshot 2023-08-17 065810

3 List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020
Screenshot 2023-08-17 065819

4 What animal has the most visits to vets?
Screenshot 2023-08-17 065827

5 Who was Maisy Smith's first visit?
Screenshot 2023-08-17 065845

6 Details for the most recent visit: animal information, vet information, and date of visit.
Screenshot 2023-08-17 065854

7 How many visits were with a vet that did not specialize in that animal's species?
Screenshot 2023-08-17 065907

8 What specialty should Maisy Smith consider getting? Look for the species she gets the most.
Screenshot 2023-08-17 065919

9 Who was the last animal seen by William Tatcher?
Screenshot 2023-08-17 065723

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -92,6 +92,72 @@ GROUP BY owners.full_name
...
+FROM vets
+LEFT JOIN specializations ON vets.id = specializations.vet_id
+LEFT JOIN species ON specializations.species_id = species.id;
+
+SELECT animals.name AS animal_name
+FROM animals
+JOIN visits ON animals.id = visits.animal_id
+JOIN vets ON visits.vet_id = vets.id
+WHERE vets.name = 'Stephanie Mendez'
+    AND visits.visit_date BETWEEN '2020-04-01' AND '2020-08-30';
```
Comment (1GORDON) [inline queries.sql:118]: Great job @ticoniq 🤝

- [ ] You have done well implementing the requested changes. However, I can tell you that you named the third screenshot with the wrong question. Kindly rename this 👇 properly. It should be `List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020`.
query

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @ticoniq ✋

Good job so far!

## Highlights!
- Created vets table ✔️ 
- Created visits table✔️
- Created specialization table ✔️
- Added results screenshots of queries to the PR ✔️
- Used JOIN to create queries. ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 1 above)
Comment (ticoniq) [reply queries.sql:118]: Thanks for the feedback but this is not enough reason to reject the project because of the name

**Review by danmainah — APPROVED** [review]
Hi @ticoniq ,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️  Implemented change requested to ensure all screenshots are added correctly.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**